python/graphstorm/gsf.py

```diff
@@ -990,6 +990,14 @@ def set_encoder(model, g, config, train_task):
     # Set GNN encoders
     dropout = config.dropout if train_task else 0
     out_emb_size = config.out_emb_size if config.out_emb_size else config.hidden_size
+
+    # Check use edge feature and GNN encoder capacity
+    assert (config.edge_feat_name is None) or \
```

Comment:
Not exactly. Although gnn_encoder will check this, but all inheritances, e.g. RGAT, HGT, Sage, and GAT/v2 need to add the two edge feature arguments in `init` function to allow gnn_encoder to work.

If we add the arguments, these APIs are changed and users will see the new arguments. But none of them implement the support of edge features. This will confuse users.

So, I choose the current logic to control our own CLI pipeline.